### ISSUE TITLE
Feature: Have systemctl start|stop|restart|reload provide feedback via log output

### ISSUE THREAD

**Issue by iio7** [systemd/systemd#1287]
I think it would be nice to have systemctl start|stop default provide feedback about whether a service has been started or stopped.

That is like issuing both these commands in one go:

systemctl start foo
systemctl status foo

Currently no feedback is provided and you manually have to check whether the service had been "started" or "stopped" afterwards.

Just an idea.

Kindest regards.

**Comment by filbranden** [comment]
I think this is similar to this [TODO](https://github.com/systemd/systemd/blob/master/TODO#L253) item:

```
* add "systemctl start -v foobar.service" that shows logs of a service
  while the start command runs. This is non-trivial to do without
  races though, since we should flush out all journal messages before
  returning from the "systemctl stop".
```

**Comment by beetlejesss** [comment]
Bump?

**Comment by poettering** [comment]
@beetlejesss this so far has not been a high-priority item for us. The best way to ensure this is implemented soon is by prepping a patch and submitting it as PR.

**Comment by greencopper** [comment]
I don't think this issue should be implemented as it's just a matter of exporting a bash function:

```
function sys-start {
    systemctl start "$1";
    systemctl status "$1";
}

export -f sys-start

function sys-stop {
    systemctl stop "$1";
    systemctl status "$1";
}

export -f sys-stop
```

Or something similar.

**Comment by phemmer** [comment]
@greencopper That may be close to what the original request is, but it's very different from the TODO item mentioned up above, which is much more powerful, but would still address the original use case.

**Comment by greencopper** [comment]
@phemmer I don't find the point in the TODO that much different from the above, one could just change the "systemctl status" to "systemctl status -f" and tail the log and then press CTRL-C to quit the tail.

Of course having the functionality build-in is nice, but I don't think it's worth the extra work.

**Comment by tuxayo** [comment]
Workaround:

`journalctl -f -u my-service & systemctl start my-service.service`

**Comment by poettering** [comment]
I think we have most of the infrastructure in place now to make this workable and safe. I would propose the following:


1. Teach journalctl mode where on receiving SIGTERM/SIGINT it would asynchronously issue the journald `io.systemd.Journal.Synchronize()` varlink call and continue log display until that call completes. journalctl already has `--sync` wrapping the call, so the code is already there somewhat (but synchronous, not asynchronous so far). 
2. Teach systemctl to optionally fork off journalctl with the right parameters, similar to how we fork off the ask-password and polkit agents. We'd kill journalctl again before we exit, synchronously, exactly like we do it for the two agents.

This makes things race-free as we always wait for operations systemctl requests from PID1 to complete, and if we then synchronously wait for all logs that have been generated up to that point to be dispatched (which is what the varlink call does), then we'll not miss a single log message related to the operation we execute.

**Comment by poettering** [comment]
Hmm, thinking about this more, we also need:

3. Teach journalctl some form of ready notification (i.e. via `sd_notify(READY=1)` or so), so that systemctl can wait until it successfully established its journal watches. Otherwise we might issue the PID 1 operation before we know the log output started.

**Comment by Hi-Angel** [comment]
> Workaround:
> 
> `journalctl -f -u my-service & systemctl start my-service.service`

Doesn't work for me. I'm debugging why `systemctl start foo.target` hangs, and the target itself produces no output, so I can't do a `journalctl -f -u foo.target`. The target though does start dozens of services, but they produce lots of output, so looking at just `journalctl -f` doesn't help either.

**Comment by Hi-Angel** [comment]
> > Workaround:
> > `journalctl -f -u my-service & systemctl start my-service.service`
> 
> Doesn't work for me. I'm debugging why `systemctl start foo.target` hangs, and the target itself produces no output, so I can't do a `journalctl -f -u foo.target`. The target though does start dozens of services, but they produce lots of output, so looking at just `journalctl -f` doesn't help either.

Okay, I just found here, it turns out systemd does actually log what it does, it's just that it prints that into journal log rather than on stdout. In case someone else as well was confused by the existence of this issue: to look at what `systemctl start foo` does you first have to execute `journalctl -f -t systemd` and then execute the `systemctl start foo`. You can also control log levels `systemd-analyze log-level`

**Comment by snwflake** [comment]
any news?

**Comment by poettering** [comment]
Closing because of c8aff47a9cebbc726fc851602612cc52710eef43
